doc/benchexec-in-container.md

```diff
@@ -0,0 +1,233 @@
+<!--
+This file is part of BenchExec, a framework for reliable benchmarking:
+https://github.com/sosy-lab/benchexec
+
+SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+# BenchExec: Containerized Environments
+
+This document explains the complications of using BenchExec inside
+containerized environments (such as Docker or Podman) and shows how you can
+create your own interactive Docker image (or adapt existing ones) to use
+BenchExec within it.
+
+Note: It is possible to use BenchExec inside other container environments, e.g.
+Docker, but we strongly recommend to use [Podman](https://podman.io/)
+(compatible with Docker), as it provides "rootless" containers, i.e. its
+containers are started as a regular user without sudo (just like BenchExec
+containers). However, as the setup is largely the same, we provide it for
+either case. In case you do want to use Docker, please also consider following
+the (easy to implement) [security recommendations](#securing-docker-execution).
+
+If you only require a non-interactive setup, i.e. only run a single BenchExec
+process inside a container, you can avoid most of the complications. See
+[below](#non-interactive-setups) for more details.
```

Comment:
How about handling the easy case first? Basically just move that section up here and adjust the glue text. It would make it more clear that it is possible to run BenchExec inside a container without lots of configuration at least in some cases. Otherwise people might see the length of the instructions and just give up.

Reply:
Hmmm, yes, lets have a look.